Assets/Scripts/Framework/Gameplay/Interactions/Interactable.cs

```diff
@@ -0,0 +1,58 @@
+using System.Collections;
+using FishNet.Observing;
+using FishNet.Object;
+using UnityEngine;
+using System;
+
+using DerailedDeliveries.Framework.Gameplay.Player;
+
+namespace DerailedDeliveries.Framework.Gameplay.Interactions
+{
+    /// <summary>
+    /// A class that is responsible for holding values for an interactable object.
+    /// </summary>
+    [RequireComponent(typeof(NetworkObject), typeof(NetworkObserver))]
+    public class Interactable : NetworkBehaviour
+    {
+        /// <summary>
+        /// Gets called when a player interacts with this interactable.
+        /// </summary>
+        public Action OnInteract;
+
+        [SerializeField]
+        private float _cooldown = .5f;
+
+        private protected bool IsOnCooldown { get; set; }
+
+        private protected bool IsInteractable { get; set; } = true;
+
+        private protected bool IsBeingInteracted { get; set; }
```

Comment:
`IsBeingInteracted` is only used and set in `Grabbable` so I think it is better to move it to that script.